### PR TITLE
Feat/public private route

### PR DESCRIPTION
**Pull request by Azim-941223**
Issue #:
[Trello](https://trello.com/c/dO4rB4Oi)

## PR type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes (please describe):
- [x] No

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Accessibility support has been tested
- [ ] Responsive layout and functionality have been tested
  - [ ] desktop and mobile view responsiveness are supported
  - [ ] no loss of functionality or content when zoomed in to 400%
  - [ ] no horizontal scrolling when zoomed in to 400% (only vertical scrolling, exception: DataGrids)
- [ ] avoids using many levels of prop drilling
- [ ] check console on errors and warnings.
---

References:
- [Clean code](https://github.com/airbnb/javascript)

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -8179,6 +8179,8 @@ redent@^3.0.0:
     indent-string "^4.0.0"
     strip-indent "^3.0.0"
 
+<<<<<<< HEAD
+=======
```
Comment (Amanbek22) [inline yarn.lock]: please remove this lines

File: yarn.lock [hunk 2]
```diff
@@ -8191,6 +8193,7 @@ redux@^4.1.2:
   dependencies:
     "@babel/runtime" "^7.9.2"
 
+>>>>>>> develop
```
Comment (Amanbek22) [inline yarn.lock]: this too

File: yarn.lock (hunk 1 above)
Comment (Azim-941223) [reply yarn.lock]: Done

File: yarn.lock (hunk 2 above)
Comment (Azim-941223) [reply yarn.lock]: Done

**Review by Bezymeys — APPROVED** [review]

**Pull request merged**